app/lib/features/activities/widgets/space_activities_section/item_widgets/type_widgets/eventDateChange.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:acter/features/activities/widgets/space_activities_section/item_widgets/activity_item_container_widgets.dart';
+import 'package:acter/features/events/utils/events_utils.dart';
+import 'package:acter/l10n/generated/l10n.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+
+class ActivityEventDateChangeItemWidget extends StatelessWidget {
+  final Activity activity;
+  const ActivityEventDateChangeItemWidget({super.key, required this.activity});
+
+  @override
+  Widget build(BuildContext context) {
+    final activityObject = activity.object();
+    // Get the new date (if available)
+    final newDate = activity.newDate();
+
+    // Handle the case when newDate is null
+    final startDate = newDate != null ? getDateFormat(newDate) : null;
+    final startTime = newDate != null ? getTimeFromDate(context, newDate) : null;
+
+    // Use a fallback message if newDate is null
+    final dateText = newDate != null
+        ? '$startDate - $startTime'
+        : ''; // Assuming noDateAvailable is a string in your localization.
```

Comment:
Why to check `null` condition multiple time for the same variable?

```suggestion
    final activityObject = activity.object();
    final newDate = activity.newDate();
    
    String? newEventSchedule;
    if(newDate!=null){
    final startDate = getDateFormat(newDate);
    final startTime = getTimeFromDate(context, newDate);
    newEventSchedule = '$startDate - $startTime';
    }
```